clang/lib/CConv/Constraints.cpp

```diff
@@ -479,15 +473,13 @@ bool Constraints::graph_based_solve(ConstraintSet &Conflicts) {
     if (!res) {
       std::set<VarAtom *> rest;
       env.doCheckedSolve(true);
-      for (const auto &C : Conflicts) {
-        if (Geq *geq = dyn_cast<Geq>(C)) {
-          VarAtom *VA = dyn_cast<VarAtom>(geq->getLHS());
-          if (!VA)
-            VA = dyn_cast<VarAtom>(geq->getRHS());
-          assert(VA != nullptr);
-          env.assign(VA, getWild());
-          rest.insert(VA);
-        }
+      for (VarAtom *VA : Conflicts) {
+        assert(VA != nullptr);
+        std::string Rsn = "Bad pointer type solution";
+        Geq *ConflictConstraint = createGeq(VA, getWild(), Rsn);
+        addConstraint(ConflictConstraint);
+        SolChkCG.addConstraint(ConflictConstraint, *this);
+        rest.insert(VA);
       }
       Conflicts.clear();
       /* FIXME: Should we propagate the old res? */
```

Comment:
Does the above `FIXME` still apply? I.e., you are basically doing what it says?

Reply:
Not sure.

Reply:
Thought about this some more. `res` is a boolean that indicates if the constraint solver was able to find a consistent solution. False means there was some contradiction. At this point in the code, we've detected a contradiction, and attempted to fix it by constraining more atoms to WILD. The question asked in the FIXME is if the final `res` value (which is the functions return) should indicate that solving initially failed for pointer type constraints, even if we fixed the solution.

This doesn't matter at all because the return value from `graph_based_solve` is never used.

Reply:
OK, then I would drop it.